lib/ddtrace/sampler.rb

```diff
@@ -247,7 +248,15 @@ def priority_assigned_upstream?(span)
 
     def priority_sample!(span)
       preserving_sampling(span) do
-        @priority_sampler.sample!(span)
+        @priority_sampler.sample!(span).tap do
+          # We want to make sure the span is tagged with the agent-derived
+          # service rate. Retrieve this from the rate by service sampler.
+          # Only do this if it was set by a RateByServiceSampler.
+          if @priority_sampler.is_a?(RateByServiceSampler)
```

Comment:
The problem we might have here is that if the `@priority_sampler` is a `RuleSampler` (new default https://github.com/DataDog/dd-trace-rb/blob/930d8210ab2bc6f54b35503aa41be9bb674c1c93/lib/ddtrace/tracer.rb#L409) then we still need to set this tag when the `RuleSampler` uses a `RateByServiceSampler`